packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -0,0 +1,157 @@
+import fs from 'fs';
+import path from 'path';
+import { AxiosError } from 'axios';
+import * as crypto from 'node:crypto';
+import { apiConfig } from '../config/api';
+import { compressFilesToTar } from '../utils/compress-files';
+import createProjectAction from '../create-project/action';
+import type { CLIContext, ProjectInfos } from '../types';
+import { getTmpStoragePath } from '../config/local';
+import { cloudApiFactory, tokenServiceFactory, local } from '../services';
+import { notificationServiceFactory } from '../services/notification';
+import { loadPkg } from '../utils/pkg';
+import { buildLogsServiceFactory } from '../services/build-logs';
+
+const FILE_SIZE_LIMIT = 100 * 1024 * 1024; // 100MB
```

Comment:
should this only be handled on the server side so it can be dynamic depending on our infra capabilities ?